Kernel/Devices/Storage/StorageDevice.cpp

```diff
@@ -144,17 +148,16 @@ ErrorOr<size_t> StorageDevice::read(OpenFileDescription&, u64 offset, UserOrKern
     return pos + remaining;
 }
 
-bool StorageDevice::can_read(OpenFileDescription const&, u64 offset) const
-{
-    return offset < (max_addressable_block() * block_size());
-}
-
 ErrorOr<size_t> StorageDevice::write(OpenFileDescription&, u64 offset, UserOrKernelBuffer const& inbuf, size_t len)
 {
+    // NOTE: The last available offset is actually just after the last addressable block.
+    if (offset >= (max_addressable_block() + 1) * block_size())
+        return 0;
```

Comment:
I think we should have a more descriptive error return here. If we try to write past a block device, Linux returns ENOSPC. We could do something like that here. 

For reading, it is fine if we return 0 as don't do anything.